to.etc.domui/src/main/java/to/etc/domui/component/buttons/ActionButton.java

```diff
@@ -0,0 +1,68 @@
+package to.etc.domui.component.buttons;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import to.etc.domui.component.layout.FloatingDiv;
+import to.etc.domui.component.menu.IUIAction;
+import to.etc.domui.component.misc.Icon;
+import to.etc.domui.component2.popupmenus.PopupMenu2;
+import to.etc.domui.dom.html.HR;
+import to.etc.domui.dom.html.NodeBase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Button with additional actions.
+ * Additional actions are activated using side button, and show internal popup menu.
+ */
+@NonNullByDefault
+public class ActionButton<T> extends DefaultButton {
+
+	private final List<IUIAction<T>> m_actions = new ArrayList<>();
+
+	private final T m_instance;
+
+	public ActionButton(T instance, IUIAction<T> action) throws Exception {
+		super(action);
+		m_instance = instance;
+	}
+
+	public ActionButton<T> addAction(IUIAction<T> action) {
```

Comment:
Why would all instances need to be a "T"? I think both the primary action and the additional actions could have different instances?

Reply:
fixed